streamparse/storm/bolt.py

```diff
@@ -446,32 +403,44 @@ def _batch_entry_run(self):
                 # later batch raises an exception
                 self._batches[key] = []
             self._batches = defaultdict(list)
+            self._tick_counter = 0
 
-    def _batch_entry(self):
-        """Entry point for the batcher thread."""
-        try:
-            while True:
-                self._batch_entry_run()
-        except Exception as e:
-            log_msg = ("Exception in {}.run() while processing tuple batch "
-                       "{!r}.".format(self.__class__.__name__,
-                                      self._current_tups))
-            log.error(log_msg, exc_info=True)
-            self.raise_exception(e, self._current_tups)
+    def process(self, tup):
```

Comment:
I had initially called it `process` because I hoped to not have to override `_run` here and just have `BatchingBolt` simply be like any other subclass of `Bolt`, only we provide it.  In the end I had to scrap that because of the auto-acking in `Bolt._run`.

That said, if we call it something else, if people accidentally implement `process`, that method will never actually be called, whereas here it would break everything because nothing will ever get added to batches.  I'm not sure which would be easier for someone to spot.